header container - "Github component, Header component, Menu componen…

tatevgrigoryan opened the pull request:
…t, Section component"

Commit:
header container - "Github component, Header component, Menu component, Section component"

arturvardanyan commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Տաթև ջան այս ֆայլը ջնջի, ավելացրու .gitignore ֆայլում։

arturvardanyan commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,27 @@
+import PropTypes from 'prop-types';
+import Section from '../Section/Section.js';
+
+
```
Տաթև ջան դատարկ տող։

anisargsian commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+import './Menu.css'
+
+
+
```
Ավելորդ դատարկ տողեր կան Տաթև ջան

anisargsian commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+);
+
+
+
```
դատարկ տողեր

anisargsian commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+Menu.PropTypes = {
+    item1:PropTypes.string.isRequired,
+    item2:PropTypes.string.isRequired,
+};
```
Դինամիկ սարքելուց հետո proptypes-երը  համապատասխանեցնել։

anisargsian commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+
+export default Menu;
+
+
```
դատարկ տողեր

anisargsian commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+        </a>
+    </ul>
+);
+
```
Դարձնել ունիվերսալ, որպեսզի մենյուի իթեմների քանակը փոխելիս մենյուն դինամիկ փոխվի

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+import './GitHub.css'
+
+
+
```
ավելորդ դատարկ տողեր

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+const GitHub = ({text}) => (
+
+    <div className="github-logo">
+        <a href="https://github.com/" target="_blank">
```
Href-ում դնել պռոյեկտի ռեպոյի լինքը

arturvardanyan replied:
Href-ը թող դրսից ստանա, text-ի փոխարեն: text-ը թող ստատիկ մնա GitHub:

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+    <div className="github-logo">
+        <a href="https://github.com/" target="_blank">
+            <span>{text}</span>
+            <img src={logo} alt={logo}/>
```
alt-մեջ պետք է լինի ինչ-որ տեքստ կամ դատարկ սթրինգ

arturvardanyan replied:
<img>-ի փոխարեն GitHubLogo կոմպոնենտ սարքի, որը SVG կվերադարձնի ու հնարավոր կլինի դրսից className փոխանցել, որ :hover-ի ժամանակ գույնը փոխենք Logo-ի:

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+            <span>{text}</span>
+            <img src={logo} alt={logo}/>
+        </a>
+
```
դատարկ տող

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+);
+
+
+
```
ավելորդ դատարկ տողեր

anisargsian commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+
+export default GitHub;
+
+
```
էստեղ էլ դատարկ տողեր

arturvardanyan commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,27 @@
+
+
+
+const Menu = ({item1, item2}) => (
```
Տաթև ջան destructuring անելուց բացվող ձևավոր փակագցից հետո ու փակվող ձևավոր փակագցից առաջ space դիր։ ... ({ item1, item2 }) => ....

anisargsian commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,27 @@
+            <div>
+              <Section />
+
+            </div>
```
առաջարկում եմ Section կոմպոնենտը ջնջել, դիվ-ի փոխարենը դնել հենց header էլեմենտը և  Section -ի պարունակությունը դնել անմիջապես հեդեռի մեջ

anisargsian commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,27 @@
+Header.propTypes = {
+    children: PropTypes.node.isRequired,
+    ClassName:PropTypes.string.isRequired,
+}
```
եթե վերը նշված փոփոխությունները կատարվում են, ապա propTypеs-երը անհրաժեշտ է համապատասխանեցնել,  այժմ header կոմպոնենտը props եր չունի , հետևաբար propTypes գրելը անիմաստ է

anisargsian commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,27 @@
+
+export default Header;
+
+
```
դատարկ տողեր

arturvardanyan commented on src/components/Github/GitHub.js:
```diff
@@ -0,0 +1,28 @@
+
+
+GitHub.PropTypes = {
+    text:PropTypes.string.isRequired,
```
Բացատներին խնդրում եմ ուշադրություն դարձրու: «:»-ից հետո մի հատ space:

arturvardanyan commented on src/components/Header/Header.css:
```diff
@@ -0,0 +1,4 @@
+body {
+    padding: 0;
+    margin: 0;
+}
```
Սա այստեղ պետք չի զրոյացնել: Դրսում մի հատ App.css պետք է ունենանք/ունենք, որը սա կանի:

arturvardanyan commented on src/components/Section/Section.js:
```diff
@@ -0,0 +1,20 @@
+
+export default Section;
+
+
```
Այս ֆայլը ավելորդ է: Վերևում նշված կառուցվածով փոխիր:

Commit:
1. change components structure

Commit:
1. improved components structure

Commit:
1. improved components structure
2. add .idea file in .gitignore
3. change Menu component to dinamic component
4. style fixing
